spec.bs

```diff
@@ -770,16 +835,31 @@ Second, at the end of the same nested scope, add the following step:
 </div>
 
 The {{WindowSharedStorage}}'s {{WindowSharedStorage/selectURL()}} method steps
-are modified in two ways. First, add the following step in the nested scope just
-after "Let |operation| be |operationMap|[|name|]." (renumbering later steps as
-appropriate):
+are modified in three ways. First, add the following steps just after step 5
+("If {{Worklet/addModule()}} has not yet been called, ..."), renumbering later
+steps:
 <div algorithm="shared-storage-selecturl-monkey-patch-1">
-2. Let <var ignore>batchingScope</var> be a new [=batching scope=].
-1. Let <var ignore>debugScope</var> be a new [=debug scope=].
+6. Let |contextId| be null.
```

Comment:
Could we define an algorithm for these steps and share in both paths?

Reply:
Had a go -- my version still needs two steps in each algorithm, but definitely less repetition